src/auth/auth.ts

```diff
@@ -354,7 +354,7 @@ class AuthService {
 		const addressDoc = await firebaseAdmin.firestore().collection('addresses').doc(address).get();
 		if (addressDoc.exists) throw apiErrorWithStatusCode(messages.ADDRESS_SIGNUP_ALREADY_EXISTS, 400);
 
-		const username = uuidv4().split('-').join('').substring(0, 25);
+		const username = address;

```

Comment:
as discussed on the last call, we'll have to keep using uuid.
and add a flag for custom_username: boolean

Reply:
will fix

Reply:
fixed